survey_formio/models/survey_question.py

```diff
@@ -0,0 +1,216 @@
+# Copyright 2023 Acsone SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+
+from odoo import models
+
+ODOO_TO_FORM_IO_TYPES_DICT = {
+    "free_text": "textarea",
+    "textbox": "textfield",
+    "numerical_box": "number",
+    "date": "day",
+    "datetime": "datetime",
+    "simple_choice": "radio",
+    "multiple_choice": "selectboxes",
+}
+
+
+class SurveyQuestion(models.Model):
+    _inherit = "survey.question"
+
+    def _get_common_parameters(self):
+        """
+        Returns a dict representing the common parameters for all form.io components.
+
+        Form.io parameters:
+        - type : required
+        - key : required
+        - input : required but no corresponding parameter in survey.
+                  Set to default form.io value
+        - label
+        - validate : only constr_mandatory supported at the moment,
+                     will be handled in its own function
+
+        See https://github.com/formio/formio.js/wiki/Components-JSON-Schema#common-parameters
+        for more information
+        """
+        self.ensure_one()
+        return {
+            "type": ODOO_TO_FORM_IO_TYPES_DICT[self.question_type],
+            "key": self.id,
```

Comment:
```suggestion
            "key": f"q{self.id},
```

It may be better to prefix these keys, so that are not int. It's a bit unusual to have integer as field names in a form, and may break things downstream.

Reply:
Done